diff --git a/src/grep.c b/src/grep.c
index ac9891d..8dbf86e 100644
--- a/src/grep.c
+++ b/src/grep.c
@@ -49,7 +49,7 @@
 #include "xalloc.h"
 #include "xstrtol.h"
 
-#define SEP_CHAR_SELECTED '+'
+#define SEP_CHAR_SELECTED ':'
 #define SEP_CHAR_REJECTED '-'
 #define SEP_STR_GROUP    "--"
 
@@ -937,10 +937,7 @@ print_line_head (char const *beg, char const *lim, char sep)
           lastnl = lim;
         }
       if (pending_sep)
-	  {
-        print_sep (' ');
         print_sep (sep);
-	  }
       print_offset (totalnl, 4, line_num_color);
       pending_sep = true;
     }
@@ -950,10 +947,7 @@ print_line_head (char const *beg, char const *lim, char sep)
       uintmax_t pos = add_count (totalcc, beg - bufbeg);
       pos = dossified_pos (pos);
       if (pending_sep)
-	  {
-        print_sep (' ');
         print_sep (sep);
-	  }
       print_offset (pos, 6, byte_num_color);
       pending_sep = true;
     }
@@ -967,9 +961,7 @@ print_line_head (char const *beg, char const *lim, char sep)
       if (align_tabs)
         fputs ("\t\b", stdout);
 
-      print_sep (' ');
-      print_sep (' ');
-      print_sep (' ');
+      print_sep (sep);
     }
 }
 
@@ -1007,7 +999,7 @@ print_line_middle (const char *beg, const char *lim,
              but is it selected or rejected/context?  */
           if (only_matching)
             print_line_head (b, lim, (out_invert ? SEP_CHAR_REJECTED
-                                      : ' '));
+                                      : SEP_CHAR_SELECTED));
           else
             {
               pr_sgr_start (line_color);
@@ -1655,10 +1647,7 @@ grepdesc (int desc, bool command_line)
             {
               print_filename ();
               if (filename_mask)
-			  {
-                print_sep (' ');
                 print_sep (SEP_CHAR_SELECTED);
-			  }
               else
                 fputc (0, stdout);
             }
